### PR TITLE
installer: add uninstall subcommand in Rust

### PR DESCRIPTION
**Pull request by kevinkassimo**
Continuation of #3804 , now in Rust.

### CODE REVIEW

File: cli/installer.rs [hunk 1]
```diff
@@ -94,6 +105,7 @@ fn generate_executable_file(
   // TODO: remove conditionals in generated scripts
   let template = format!(
     r#"#!/bin/sh
```
Comment (bartlomieju) [inline cli/installer.rs]: While you at it, could you remove those conditionals in generated scripts?

Comment (kevinkassimo) [reply cli/installer.rs]: Do you mean the branch of `$basedir/deno` and similar since deno binary have been moved to `.local`?

Comment (bartlomieju) [reply cli/installer.rs]: Yeah, the "if" branches in both generated scripts - I don't really know what it was for but @ry wants it gone

**Comment by ry** [comment]
It's a little annoying to have a whole separate subcommand for a niche functionality related to "deno install". I wonder if "deno install --uninstall" wouldn't be better? That doesn't read very nicely, but I think it makes more sense.

We could rename "deno install" to something else. Maybe "deno bookmark". Then you would have "deno bookmark https://foo/bar.js" and "deno bookmark --delete https://foo/bar.js" ? Not sure about that either. Just thinking outloud.

**Comment by kevinkassimo** [comment]
@ry IMHO to have a flag that reverts the behavior of a command is just SO confusing for the users...

Also having a separate command, we can have some clearer behavior like uninstalling multiple command if provided more than one argument.

The flag makes sense under more generic subcommand names like `bookmark`, though I also feel `bookmark` like a very strange name -- what we generate is indeed a command on its own: can by typed and run write away. `bookmark` feels more like caching to me, which we already do

**Comment by nayeemrmn** [comment]
I've always thought `deno install` was awkward for what it does, especially when you think about its behaviour on a local module. It definitely acts more like a `bookmark`, or [`alias`](https://github.com/denoland/deno/issues/3800#issue-555291367), besides the tacked-on prefetch.

**Comment by cknight** [comment]
I'd agree that `deno install` is confusing.  I was very surprised the first time I had a look under the covers at what the deno install actually did.  Install to me means placing a binary on my system for use, not creating a run script for source code.  I think `install` should be renamed.

**Comment by kevinkassimo** [comment]
How about just `deno command` or `deno cmd` (since we are installing commands)? In that case a delete flag `-d` makes sense.
+ `deno command -i ...` for adding new commands
+ `deno command -d ...` for deleting existing commands
+ `deno command -l` for listing all installed commands

I also have problem with `alias` since it does feels like creating symlinks or local aliases instead of a command that can be invoked anywhere

**Comment by ry** [comment]
It would be cool to use "deno bundle" to install programs. On mac/linux we can use shebang to eliminate the shell script entirely.
On Windows we'd have to find somewhere to store the bundle (obvious choice would be as a sibling of the .cmd).

**Comment by kevinkassimo** [comment]
@ry I am actually thinking about something like `deno install -b ...` which would install the file as bundled (I prefer this over adding flags to `deno bundle`, since its functionality is closer to `install`, and the other flags currently on `install` can be used here)

A minor problem with shebang is that while `env -S` allows multiple arguments, we cannot store the extra arguments/flags that will be accepted by the script itself (instead of by Deno).

Maybe I can create a subdirectory under installation direction to save all the bundled files there, and still use a basic shell/batch wrapper to invoke them.

**Comment by bartlomieju** [comment]
Thanks for the patch @kevinkassimo but seems it superfluous to have another subcommand/flag for something so trivial when you can just `rm` it

**Pull request closed without merging**